### ISSUE TITLE
Commit failure in baseline tests

### ISSUE THREAD

**Issue by TomSweeneyRedHat** [projectatomic/libpod#692]
**Description**
When I ran the ./test/test_podman_baseline.sh on a new Fedora 27 vm with the latest podman, I ran into errors when running containers that had been committed to.

**Steps to reproduce the issue:**

1.  Run '/bin/bash -v ./test/test_podman_baseline.sh' or the following portion of the script:

```
########
# Create Fedora based container
########
image=$(podman pull fedora)
echo $image
podman run $image ls /

########
# Create shell script to test on
########
FILE=./runecho.sh
/bin/cat <<EOM >$FILE
#!/bin/bash
for i in {1..9};
do
    echo "This is a new container pull ipbabble [" \$i "]"
done
EOM
chmod +x $FILE

########
# Copy and run file on container
########
ctrid=$(podman ps --all -q)
mnt=$(podman mount $ctrid)
cp ./runecho.sh ${mnt}/tmp/runecho.sh
podman umount $ctrid
podman commit $ctrid runecho
podman run runecho ./tmp/runecho.sh
```

**Describe the results you received:**

No errors noted prior to the run command.
```
# podman run runecho ./tmp/runecho.sh
container create failed: container_linux.go:348: starting container process caused "process_linux.go:402: container init caused \"rootfs_linux.go:58: mounting \\\"/var/lib/containers/storage/overlay-containers/f69978b5052805d458956faff88103dadacdf4f31faff5a49977ea9579b132c8/userdata/volumes/proc\\\" to rootfs \\\"/var/lib/containers/storage/overlay/2985c424e19dbfd588ce1513ff6244074958de2a45cc4f04626e51ad1451bfa7/merged\\\" at \\\"proc\\\" caused \\\"\\\\\\\"/var/lib/containers/storage/overlay/2985c424e19dbfd588ce1513ff6244074958de2a45cc4f04626e51ad1451bfa7/merged/proc\\\\\\\" cannot be mounted because it is located inside \\\\\\\"/proc\\\\\\\"\\\"\""
: internal libpod error

```

**Describe the results you expected:**

The container should run showing with no errors and a number of lines of output from a counter.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
# podman version
Version:       0.4.4
Go Version:    go1.9.5
OS/Arch:       linux/amd64

# ls  /tmp/p*.rpm
podman-0.4.4-1.git888927a.fc27.x86_64.rpm
```

**Output of `podman info`:**

```
# podman info
host:
  MemFree: 909926400
  MemTotal: 2089639936
  SwapFree: 2141188096
  SwapTotal: 2147479552
  arch: amd64
  cpus: 1
  hostname: localhost.localdomain
  kernel: 4.15.10-300.fc27.x86_64
  os: linux
  uptime: 1h 40m 45.68s (Approximately 0.04 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - registry.access.redhat.com
store:
  ContainerStore:
    number: 2
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 3
  RunRoot: /var/run/containers/storage

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Virtual Machine on my PC.  Fedora 27 freshly installed.  I did `dnf -y update` to get it up to date and then also `dnf -y docker` and also installed the podman kit that I'd pulled from the Fedora site..

**Comment by mheon** [comment]
This has to be default mounts in the spec.

https://github.com/projectatomic/libpod/blob/master/libpod/container_commit.go#L99-L101

We can't do this in libpod because libpod has no concept of a volume mount vs a default mount - it only knows about mounts in the spec.

We need to move this part into cmd/podman/ and base it off the CreateConfig artifact. If we can't find the artifact, we just commit without volumes.

**Comment by mheon** [comment]
For the libpod side of things, we're probably better off just basing the volumes off what's in the image we're building from. Same for CMD and ENTRYPOINT, probably - set the real ones based on CreateConfig in cmd/podman and use the image ones in libpod if not overridden.

**Comment by mheon** [comment]
To clarify my thinking here... in `commit.go` we add all the mounts in the spec we are given to the image we create. This doesn't work because the spec contains many default mounts as well - `/proc` for example - that should never make it into an image. We can't really determine what these are on the `libpod` side, so we can't use the spec mounts in Libpod to commit.

I'd really prefer to avoid adding large portions of the unparsed CLI into libpod - CRI-O is never going to use them. Therefore, I think we should move the volume mount additions into `cmd/podman /commit.go` where we can get at the original parsed command line (and its volume mounts) via the create config artifact we use for `ps` and `inspect`. If the user provides volumes for the new image at the command line, we pass those into libpod; otherwise, we pass in the volumes from the create config. On the libpod side, if not overridden, it's probably safest to just use the volumes from the image the container is based on.

**Comment by mheon** [comment]
We might want to cut a `v0.4.4.1` tag or similar once a fix is in, because I think every image we commit right now is going to be broken because of this. We should also add an integration test to run a committed image to verify that works.

**Comment by TomSweeneyRedHat** [comment]
Just realized I'd left off the required pull/run commands in the reproducer.  I've just updated my original comment with them added in.

**Comment by baude** [comment]
We cannot be looking at the image a container is based on for mounts.  We need to only look at the container being committed.  I believe the correct fix here is set a const of default mounts in libpod and in commit.go where are adding the volumes to the new image, we only add those NOT which are not defaults.

**Comment by baude** [comment]
ill propose a PR

**Comment by mheon** [comment]
Should be fixed now by #699
